apollo-federation/src/query_graph/graph_path.rs

```diff
@@ -178,11 +173,15 @@ where
     // TODO(@TylerBloom): Add in once defer is supported.
     #[serde(skip)]
     defer_on_tail: Option<DeferDirectiveArguments>,
```

Comment:
Refactors:
- Some renames here to match with `ContextCondition` and `ContextMapEntry`.